bottomless/src/wal.rs

```diff
@@ -0,0 +1,272 @@
+use crate::replicator::crc;
+use anyhow::{anyhow, Result};
+use std::io::{ErrorKind, SeekFrom};
+use std::path::Path;
+use tokio::io::{AsyncReadExt, AsyncSeekExt};
+
+const CRC_64: crc::Crc<u64> = crc::Crc::<u64>::new(&crc::CRC_64_ECMA_182);
+
+pub(crate) fn crc64(prev_crc: u64, data: &[u8]) -> u64 {
```

Comment:
To be fixed with SQLite-compatible implementation: https://github.com/libsql/libsql/blob/f83d455ae8b57426b501770d392062d4f51165af/src/wal.c#L70-L88